presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ResponseWrapper.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.controller;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public final class ResponseWrapper
+{
+    private int statusCode;
+    private Status status;
+    private String entity;
+    private MultivaluedMap<String, Object> headers;
+
+    private ResponseWrapper(Response response)
+    {
+        this.statusCode = requireNonNull(response.getStatus());
+        this.status = requireNonNull(response.getStatusInfo().toEnum());
```

Comment:
Why store both the status code and the status itself? You should either store the status code and maybe the reason phrase, or the entire status, as a `StatusType` (not a `Status`).